pkg/deploy/controller/deploymentconfig/controller.go

```diff
@@ -222,24 +241,19 @@ func (c *DeploymentConfigController) findDetails(config *deployapi.DeploymentCon
 		if _, err := c.osClient.ImageStreamTags(config.Namespace).Get(name, tag); err != nil {
 			if !errors.IsNotFound(err) {
 				glog.V(2).Infof("Error while trying to get image stream tag %q: %v", istag, err)
-				return details, existingDeployments, latestDeploymentExists, nil
+				return details, nil
 			}
 			// In case the image stream tag was not found, then it either doesn't exist or doesn't exist yet
 			// (a build configuration output points to it so it's going to be populated at some point in the
 			// future)
 			details = fmt.Sprintf("The image trigger for image stream tag %q will have no effect because image stream tag %q does not exist.", istag, istag)
-			bcList, err := c.osClient.BuildConfigs(kapi.NamespaceAll).List(labels.Everything(), fields.Everything())
-			if err != nil {
-				glog.V(2).Infof("Error while trying to list build configs: %v", err)
-				return details, existingDeployments, latestDeploymentExists, nil
-			}
-
-			for _, bc := range bcList.Items {
+			for _, obj := range c.buildConfigs.List() {
```

Comment:
Restricting to namespace is a start. I think one of the benefits of using a `cache.Store` is we have options for optimizing how the store is populated.

For example, it could be further narrowed by only filling the store with buildConfigs with field selector `spec.output.to.kind=ImageStreamTag`.

Taken even further, if `spec.output.to.name in <list>` where `<list>` is computed by collecting image stream trigger references from known deploymentConfigs, which we already list/watch and eventually will have access to via a `cache.Store` when using the controller framework. (P.S. can you use an "in" clause in a field selector?)